apps/rpc/src/modules/event/attendance-router.ts

```diff
@@ -46,34 +50,85 @@ export const attendanceRouter = t.router({
       })
     )
     .mutation(async ({ input, ctx }) => {
-      return ctx.executeTransaction(async (handle) => ctx.attendanceService.deletePool(handle, input.id))
+      return ctx.executeTransaction(async (handle) => ctx.attendanceService.deleteAttendancePool(handle, input.id))
     }),
 
   adminRegisterForEvent: staffProcedure
     .input(
       z.object({
-        attendancePoolId: AttendancePoolSchema.shape.id,
         attendanceId: AttendanceSchema.shape.id,
         userId: UserSchema.shape.id,
       })
     )
     .mutation(async ({ input, ctx }) => {
-      return ctx.executeTransaction(async (handle) =>
-        ctx.attendeeService.adminRegisterForEvent(handle, input.userId, input.attendanceId, input.attendancePoolId)
-      )
+      return ctx.executeTransaction(async (handle) => {
+        return await ctx.attendanceService.registerAttendee(handle, input.attendanceId, input.userId, {
+          ignoreRegistrationWindow: true,
+          immediateReservation: true,
+          immediatePayment: false,
+        })
+      })
+    }),
+
+  updateAttendancePayment: staffProcedure
+    .input(
+      z.object({
+        id: AttendanceSchema.shape.id,
+        price: z.number().int(),
```

Comment:
I know this wasn't your decision but why do we restrict to integers?
```suggestion
        price: z.number().int().positive(),
```